code/game/objects/items/tools/synth_bracer/synth_bracer.dm

```diff
@@ -0,0 +1,444 @@
+		if(internal_transmitter.do_not_disturb >= PHONE_DND_ON)
+			phone_status = "dnd"
+		else if(internal_transmitter.attached_to.loc != internal_transmitter)
+			phone_status = "listening"
+		else if(internal_transmitter.caller)
+			phone_status = "ringing"
+
+	var/image/phone_image = image(icon, src, "phone_[phone_status]")
+	phone_image.appearance_flags = RESET_COLOR|KEEP_APART
+
+	var/image/secondary_image = image(icon, src, "secondary_[active_utility]")
+	secondary_image.appearance_flags = RESET_COLOR|KEEP_APART
+
+	var/image/primary_image = image(icon, src, "primary_[active_ability]")
+	primary_image.appearance_flags = RESET_COLOR|KEEP_APART
+
+	if(motion_detector_active)
+		var/image/motion_image = image(icon, src, "motion_active")
+		motion_image.appearance_flags = RESET_COLOR|KEEP_APART
+		overlays += motion_image
+
+	overlays += idle_image
+	overlays += phone_image
+	overlays += secondary_image
+	overlays += primary_image
+	overlays += status_image
+
+
+/obj/item/clothing/gloves/synth/proc/check_for_ringing()
+	SIGNAL_HANDLER
+	update_overlays()
+
+/obj/item/clothing/gloves/synth/get_mob_overlay(mob/user_mob, slot)
+	var/image/overlay = ..()
+
+	if((slot != WEAR_HANDS) || !underglove)
+		return overlay
+
+	overlay = underglove.get_mob_overlay(user_mob, slot)
+
+	var/image/bracer_img = overlay_image('icons/mob/humans/onmob/synth/bracer.dmi', get_icon_state(user_mob, slot), null, RESET_COLOR|NO_CLIENT_COLOR)
+	overlay.overlays += bracer_img
+
+	return overlay
+
+/obj/item/clothing/gloves/synth/verb/swap_hands()
+	set name = "Swap Hands"
+	set src in usr
+
+	if(base_item_slot_state == "bracer_[bracer_color]")
+		base_item_slot_state = "bracer_[bracer_color]_r"
+	else
+		base_item_slot_state = "bracer_[bracer_color]"
+	item_state_slots[WEAR_HANDS] = base_item_slot_state
+
+	to_chat(usr, SPAN_NOTICE("You shift \the [src] over to your [base_item_slot_state == "bracer_[bracer_color]" ? "left arm" : "right arm"]."))
+	usr.update_inv_gloves()
+
+/obj/item/clothing/gloves/synth/verb/remove_gloves()
+	set name = "Remove Gloves"
+	set src in usr
+
+	if(!underglove)
+		return FALSE
+
+	to_chat(usr, SPAN_NOTICE("You remove [underglove] from the wrist-strap."))
+	underglove.forceMove(get_turf(usr))
+	underglove = null
+
+	usr.update_inv_gloves()
+
+//#############################
+//######## CHARGING ###########
+//#############################
+/obj/item/clothing/gloves/synth/proc/handle_apc_charge(mob/living/carbon/human/user, obj/structure/machinery/power/apc/apc)
+	if(!istype(user))
+		return FALSE
+
+	if(!(user.species.flags & IS_SYNTHETIC) || user.a_intent != INTENT_GRAB)
+		return FALSE
+
+	if(user.action_busy)
+		return FALSE
+
+	INVOKE_ASYNC(src, PROC_REF(complete_apc_charge), user, apc)
+
+/obj/item/clothing/gloves/synth/proc/complete_apc_charge(mob/living/carbon/human/user, obj/structure/machinery/power/apc/apc)
+	start_charging(user)
+
+	if(!do_after(user, 6 SECONDS, INTERRUPT_ALL, BUSY_ICON_GENERIC))
```

Comment:
You'll want to doublecheck that the user still has gloves (and optionally is within range) after the do_after